libraries/llvm/bytearray.c

```diff
@@ -164,6 +164,35 @@ struct Pos c_bytearray_equal(const struct Pos left, const struct Pos right) {
     return BooleanTrue;
 }
 
+Int c_bytearray_compare(const struct Pos left, const struct Pos right) {
+    uint64_t left_size = left.tag;
+    uint64_t right_size = right.tag;
+    
+    for (uint64_t j = 0; j < left_size && j < right_size; ++j) {
```

Comment:
Would it be worth it here to do a fast path for:
1. are the structs the same, and
2. are they the same size
?

Also, is it worth it to use `memcmp` for the comparison itself (since it should be reasonably optimised even if the compiler doesn't figure out that the for loop is a `memcmp`?)

WDYT?

Reply:
Note that all of these are also possibly a consideration for `c_bytearray_equal` -- the two functions should definitely always agree (`c_bytearray_compare == 0 <=> c_bytearray_equal == true`)...

Reply:
1. This sounds like a good idea
2. I was unsure about what the compare should prioritize. It is implemented as "c" > "abcde", so characters being the deciding factor before size. Otherwise i'll just need to switch the order of for and if, which would be the fast path

`memcmp` is probably a good idea

Reply:
Ah, yes, you're right, `2` makes only sense for the equality, not the comparison, thanks!